include/aws/common/allocator.h

```diff
@@ -228,6 +228,21 @@ size_t aws_small_block_allocator_page_size(struct aws_allocator *sba_allocator);
 AWS_COMMON_API
 size_t aws_small_block_allocator_page_size_available(struct aws_allocator *sba_allocator);
 
+/*
+ * Create an aligned allocator with customized alignment.
+ */
+AWS_COMMON_API
+struct aws_allocator *aws_customized_aligned_allocator_new(
```

Comment:
maybe just aws_custom_alignment_allocator_new?

Reply:
I assume it's named this way because the nice simple name `aws_aligned_allocator()` is already taken, but that one automagically determines the alignment, and it differs based on allocation size.

Move these new functions to be right next to the existing `aws_aligned_allocator()` (instead of the end of the file) so that someone searching "align" is more likely to see both options and pick the right one.

I like "explicit" more than "custom". "custom" sounds like "you can do whatever you want like via a callback" while "explicit" sounds to me like "we simply use this 1 number you give us".

I'd suggest `aws_explicit_aligned_allocator()` (sounds like the pre-existing allocator) or `aws_explicit_alignment_allocator()` (nicer sounding gramatically)